Update FSD: Regions and FSD: Before/After to use BAM index generated by Galaxy

natefoo opened the pull request:
Also, if `--bamIndex` is not specified, the index will be built in the current working directory rather than adjacent to `bamFile`.

Fixes #3931.

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

nsoranzo commented:
If it was me implementing this, I would not add the `--bamIndex` option but
- symlink the bam input and its index to the working directory in the Cheetah
- in the Python scripts check for the `.bai` file and create it if it doesn't exist.

But it works this way as well, of course.

natefoo commented:
@nsoranzo done. I guess I can revert to the older pysam and drop the explicit index filename handling as well.

nsoranzo commented on tools/fsd/fsd_beforevsafter.xml:
```diff
@@ -1,14 +1,18 @@
-    <command>
+    <command><![CDATA[
+#if $bamFile:
+ln -s '$bamFile' 'reads.bam' &&
+ln -s '$bamFile.metadata.bam_index' 'reads.bam.bai' &&
```
```suggestion
    ln -s '$bamFile' 'reads.bam' &&
    ln -s '$bamFile.metadata.bam_index' 'reads.bam.bai' &&
```

nsoranzo commented on tools/fsd/fsd_beforevsafter.xml:
```diff
@@ -21,7 +25,7 @@ python '$__tool_directory__/fsd_beforevsafter.py'
 #end if
```
Can you also replace

```
--bamFile '$bamFile'
```

with
```
    --bamFile reads.bam
```

just above this line please?

nsoranzo commented on tools/fsd/fsd_regions.xml:
```diff
@@ -17,7 +19,7 @@ python '$__tool_directory__/fsd_regions.py'
 #end if
```
`--bamFile reads.bam` above

nsoranzo commented:
> @nsoranzo done. I guess I can revert to the older pysam and drop the explicit index filename handling as well.

Thanks @natefoo ! I suppose the explicit index filename handling can be useful if the scripts are to be used outside of Galaxy, not sure if that's of interest.

natefoo commented:
I went ahead and reverted to pysam 0.15 and dropped the index filename syntax so that we don't generate new containers for this when it's unnecessary. If this looks good I'll squash it.

nsoranzo commented on tools/fsd/fsd_beforevsafter.xml:
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<tool id="fsd_beforevsafter" name="FSD Before/After:" version="1.0.2" profile="19.01">
+<tool id="fsd_beforevsafter" name="FSD Before/After:" version="1.0.3" profile="19.01">
```
Given that you've restored the pysam requirement version, you may also restore the tool versions since the rest are bug fixes not changing input or output. Up to you!
